core/src/test/scala/blobstore/AbstractStoreTest.scala

```diff
@@ -407,6 +408,23 @@ trait AbstractStoreTest extends AnyFlatSpec with Matchers with BeforeAndAfterAll
     result.unsafeRunSync()
   }
 
+  it should "be able to list recursively" in {
+    val dir: Path = dirPath("list-recursively")
+    val files     = List("a", "b", "c", "sub-folder/d", "sub-folder/sub-sub-folder/e").map(dir / _)
+    val result = for {
+      _     <- files.traverse(p => Stream.emit(0: Byte).through(store.put(p)).compile.drain)
+      paths <- store.list(dir, recursive = true).compile.toList
+    } yield {
+      paths must have size 5
+      paths.flatMap(_.fileName) must contain only ("a", "b", "c", "d", "e")
```

Comment:
Can you check the order here?

```
paths must contain theSameElementsInOrderAs List("a", "b", "c", "d", "e")
```

or at least that "d" appear after "a", "b", "c" and before "e"

Reply:
I don't think we want to require particular order of elements in the returned stream consistently across all stores, checking that everything is returned should be enough.